### PR TITLE
chore: shift CI/CD processes to GitHub workflows

### LINKED ISSUE #201: Move CI & CD to Github actions

### PR DESCRIPTION
**Pull request by algomaster99**
Fixes #201 

Same intent as #202 but workflows execute here. Please read [this comment](https://github.com/SpoonLabs/gumtree-spoon-ast-diff/pull/202#issuecomment-926024592) for an explaination.

### CODE REVIEW

**Comment by algomaster99** [comment]
I hope I correctly configured the deployment workflow. :crossed_fingers:

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: CI
+on: push
+
+jobs:
+  package:
```
Comment (monperrus) [inline .github/workflows/ci.yml:5]: I'd say we don't need to package in CI, the "Run test is enough"

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    if: ${{ github.repository == 'SpoonLabs/gumtree-spoon-ast-diff' }}
+    name: 'Publish package'
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Wait for tests to pass
+      uses: lewagon/wait-on-check-action@v1.1.1
+      with:
+        ref: ${{ github.ref }}
+        running-workflow-name: 'Publish package'
```
Comment (monperrus) [inline .github/workflows/publish.yml:18]: 'Publish package' is not a valid name in ci.yml. Is that a problem?

**Review by monperrus — COMMENTED** [review]
Very nice progress here, thanks a lot.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (algomaster99) [reply .github/workflows/ci.yml:5]: Oh, right. I will just compile and run tests here.

File: .github/workflows/publish.yml (hunk 2 above)
Comment (algomaster99) [reply .github/workflows/publish.yml:18]: The `wait for tests to pass` waits for the `ci.yml` to pass and only then `publish.yml` will execute on master. "Publish package" is the name of the current job so specifying that will ensure that all previous workflows are completed before publish.yml is initiated. Reference: https://github.com/lewagon/wait-on-check-action#running-workflow-name

**Comment by monperrus** [comment]
Bye bye Travis, good morning Actions!

Thanks a lot @algomaster99.

Let's see if we have a new version on Central with this merge to master.

**Comment by algomaster99** [comment]
@monperrus Oops. There are some failures. I am looking into it.

**Comment by monperrus** [comment]
thanks!

**Pull request merged**